src/main/java/com/faforever/client/chat/ChatUserService.java

```diff
@@ -42,9 +57,44 @@
   private final I18n i18n;
   private final EventBus eventBus;
 
+  private List<EmoticonsGroup> emoticonsGroups;
+  private HashMap<String, String> emoticonShortcodes;
+
   @Override
   public void afterPropertiesSet() {
     eventBus.register(this);
+    loadAndVerifyEmoticons();
+  }
+
+  private void loadAndVerifyEmoticons() {
+    try {
+      emoticonsGroups = new ObjectMapper().readValue(EMOTICONS_JSON_FILE_RESOURCE.getFile(),
+          TypeFactory.defaultInstance().constructCollectionType(List.class, EmoticonsGroup.class));
+      emoticonShortcodes = new HashMap<>();
+      List<Emoticon> emoticons = emoticonsGroups.stream().flatMap(emoticonsGroup -> emoticonsGroup.getEmoticons().stream()).collect(Collectors.toList());
+      for (Emoticon emoticon : emoticons) {
+        String shortcode = emoticon.getShortcode();
+        if (emoticonShortcodes.containsKey(shortcode)) {
+          throw new ProgrammingError("Shortcode `" + shortcode + "` is already taken");
+        }
+        emoticonShortcodes.put(shortcode, emoticon.getSvgFilePath());
```

Comment:
No need to collect to a list here if we are just going to iterate through it again. Just use forEach